File: tests/cpp/simple.cpp
#include <iostream>
#include <string>

enum Color { Red, Green, Blue };

int factorial(int n) {
    if (n == 0) {
        return 1;
    }
    return n * factorial(n - 1);
}

int main() {
    std::string msg = "hello";
    int x = 42;
    bool b = true;
    std::cout << factorial(x) << std::endl;
    return 0;
}

File: tests/zig/simple.zig
const std = @import("std");

const Color = enum {
    red,
    green,
    blue,
};

fn factorial(n: u64) u64 {
    if (n == 0) return 1;
    return n * factorial(n - 1);
}

const Point = struct {
    x: f64,
    y: f64,
};

pub fn main() void {
    const result = factorial(42);
    const msg = "hello";
    const flag = true;
    _ = result;
    _ = msg;
    _ = flag;
}
